src/sqlfluff/parser/grammar.py

```diff
@@ -463,24 +463,30 @@ def _bracket_sensitive_look_ahead_match(cls, segments, matchers, parse_context,
             return ((), MatchResult.from_unmatched(segments), None)
 
         # Get hold of the bracket matchers from the dialect, and append them
-        # to the list of matchers.
-        # TODO: Potentially have error handling here for dialects without
-        # square brackets.
-        start_brackets = [
-            parse_context.dialect.ref('StartBracketSegment'),
-            parse_context.dialect.ref('StartSquareBracketSegment')
-        ]
-        end_brackets = [
-            parse_context.dialect.ref('EndBracketSegment'),
-            parse_context.dialect.ref('EndSquareBracketSegment')
-        ]
+        # to the list of matchers. We get them from the relevant set on the
+        # dialect. We use zip twice to "unzip" them. We ignore the first
+        # argument because that's just the name.
+        _, start_bracket_refs, end_bracket_refs, definitely_bracket = zip(*parse_context.dialect.sets('bracket_pairs'))
```

Comment:
This line is pretty tricky to understand. Possible to use `namedtuple`, object, dataclass, or something like that to make the data more "self describing"?